openedx/core/djangoapps/content_tagging/rules.py

```diff
@@ -6,35 +6,169 @@
+    """
+    Return True if the given user is a content creator for any of the given orgs.
+    """
+    return len(get_content_creator_orgs(user, orgs)) > 0
+
+
+def is_org_instructor(orgs: list[Organization], user: UserType) -> bool:
+    """
+    Return True if the given user is an instructor for any of the given orgs.
+    """
+    return len(get_instructor_orgs(user, orgs)) > 0
+
+
+def get_admin_orgs(user: UserType, orgs: list[Organization]) -> list[Organization]:
+    """
+    Returns a list of orgs that the given user is an admin, from the given list of orgs.
+    """
+    return [
+        org for org in orgs if OrgStaffRole(org=org.short_name).has_user(user)
+    ]
+
+
+def get_content_creator_orgs(user: UserType, orgs: list[Organization]) -> list[Organization]:
+    """
+    Returns a list of orgs that the given user is a content creator, the given list of orgs.
+    """
+    return [
+        org for org in orgs if (
+            OrgLibraryUserRole(org=org.short_name).has_user(user) or
+            OrgInstructorRole(org=org.short_name).has_user(user) or
+            OrgContentCreatorRole(org=org.short_name).has_user(user)
+        )
+    ]
+
+
+def get_instructor_orgs(user: UserType, orgs: list[Organization]) -> list[Organization]:
+    """
+    Returns a list of orgs that the given user is an instructor, from the given list of orgs.
+    """
+    instructor_roles = CourseAccessRole.objects.filter(
+        org__in=orgs,
+        user=user,
+        role__in=(CourseStaffRole.ROLE, CourseInstructorRole.ROLE),
+    )
+    return [role.org for role in instructor_roles]
+
+
+@rules.predicate
+def can_view_taxonomy(user: UserType, taxonomy: oel_tagging.Taxonomy) -> bool:
+    """
+    Returns True if the given user can view the given taxonomy.
+
+    Taxonomy admins can view any taxonomy.
+    Org-level staff can view any taxonomy that is associated with one of their orgs.
+    Org-level course creators and instructors can view any enabled taxonomy that is owned by one of their orgs.
+    """
+    # The following code allows METHOD permission (GET) in the viewset for everyone
+    if not taxonomy:
+        return True
+
+    taxonomy = taxonomy.cast()
+
+    # Taxonomy admins can view any taxonomy
+    if oel_tagging.is_taxonomy_admin(user):
+        return True
+
+    is_all_org = TaxonomyOrg.objects.filter(
+        taxonomy=taxonomy,
+        org=None,
+        rel_type=TaxonomyOrg.RelType.OWNER,
+    ).exists()
+
+    # Only taxonomy admins can view disabled all org taxonomies
+    if is_all_org and not taxonomy.enabled:
+        return False
+
+    taxonomy_orgs = TaxonomyOrg.get_organizations(
+        taxonomy=taxonomy,
+        rel_type=TaxonomyOrg.RelType.OWNER,
+    )
+
+    # Org-level staff can view any taxonomy that is associated with one of their orgs.
+    if is_org_admin(taxonomy_orgs, user):
+
+        # All org taxonomies are enabled by default
+        return not is_all_org or taxonomy.enabled
```

Comment:
If [we handle `is_all_org` above](https://github.com/openedx/edx-platform/pull/33413/files#diff-15db056809a5257a6081c33072c7d430098eb751b7bea826c8e1325702b4c9c4R110), then can this be just this, because org-level admins can see both disabled and enable taxonomies?

```suggestion
        return True
```

Reply:
Sure! I think even if we don't change the code above this makes sense.